src/wazuh_modules/wm_sca.c

```diff
@@ -3193,3 +3201,66 @@ static int append_msg_to_vm_scat (wm_sca_t * const data, const char * const msg)
     }
     return 0;
 }
+
+/*
+    Thread that sends the integrity hash to the manager every configurable number of seconds.
+    If the manager has a different integrity hash, it will request the agent to send a new
+    sca scan and integrity hash.
+ */
+static void *wm_sca_check_integrity_periodically (wm_sca_t * data) {
+    unsigned int time_sleep = data->integrity_interval;
+
+    while (1) {
+        /* Thread sleeps for 5 minutes until the next time it sends the hash */
+        mdebug2("Sleeping for %d seconds.", (int)time_sleep);
+        wm_delay(1000 * time_sleep);
+
+        /* It doesn't send anything if there are no policies */
+        if (!data->policies) {
+            continue;
+        }
+
+        int i;
+
+        /* Send hash for every policy file */
+        for (i = 0; data->policies[i]; i++) {
+            if (!data->policies[i]->enabled) {
+                continue;
+            }
+
+            cJSON *json_integrity = cJSON_CreateObject();
+            int cis_db_index = i;
+
+            cJSON_AddStringToObject(json_integrity, "type", "integrity_check");
+            
+            mdebug1("Calculating hash for scanned results.");
+            char *integrity_hash = wm_sca_hash_integrity(cis_db_index);
+
+            /* If there is no hash in the local db, it will send an empty one */
+            if (!integrity_hash) {
+                os_realloc(integrity_hash, sizeof(""), integrity_hash);
+                strncpy(integrity_hash, "", sizeof(""));
+            }
+            
+            cJSON_AddStringToObject(json_integrity, "hash", integrity_hash);
+
+            /* If there is no policy id in the local db, it will send an empty one */
+            if(!data->policies[cis_db_index]->policy_id){
+                cJSON_AddStringToObject(json_integrity, "policy_id", "");
```

Comment:
There is a use case where the agent doesn't have the scan on start enabled where the module sends an empty hash for an empty policy_id. When the manager receives the integrity, it asks for a scan to the agent for an empty policy_id.
The agent will do nothing and this loop will continue until the first SCA scan is made.

Please, change the logic to avoid sending integrity messages for an empty policy_id.

Reply:
Fixed